extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/indexing/MSQCompactionRunner.java

```diff
@@ -159,6 +161,41 @@ public TaskStatus runCompactionTasks(
       TaskToolbox taskToolbox
   ) throws Exception
   {
+    for (Map.Entry<Interval, DataSchema> intervalDataSchema : intervalDataSchemas.entrySet()) {
+      if (Boolean.valueOf(true).equals(intervalDataSchema.getValue().getHasRolledUpSegments())) {
+        for (AggregatorFactory aggregatorFactory : intervalDataSchema.getValue().getAggregators()) {
+          // Don't proceed if either:
```

Comment:
This whole check should be performed as part of the `validateCompactionTask()` invocation.
If needed, you can update the signature of `validateCompactionTask()` to accept the data schemas.